src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -199,22 +199,51 @@ Task IHttpResponseBodyFeature.SendFileAsync(string path, long offset, long? coun
 
         // TODO: In the future this could complete the body all the way down to the server. For now it just ensures
         // any unflushed data gets flushed.
-        protected async Task CompleteResponseBodyAsync()
+        protected Task CompleteResponseBodyAsync()
         {
             if (ResponsePipeWrapper != null)
             {
-                await ResponsePipeWrapper.CompleteAsync();
+                var completeAsyncTask = ResponsePipeWrapper.CompleteAsync();
+                if (!completeAsyncTask.IsCompletedSuccessfully)
```

Comment:
If the ValueTask is completed I believe you still want to call `.GetAwaiter().GetResult();`, e.g.

https://github.com/dotnet/aspnetcore/blob/3509323ad182f47ea19cc20f5cfd5735eb447720/src/Shared/ValueTaskExtensions/ValueTaskExtensions.cs#L18-L19